### PR TITLE
Minor feeds fixes to be W3C-validator compliant

### LINKED ISSUE #914: RSS feed validation
Hello.

I checked Shaarli RSS stream with the W3C validor by curioisty, and I got 2 minor warnings:
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fchezsoi.org%2Fshaarli%2F%3Fdo%3Datom
- self reference doesn't match document location: because of HTTP vs HTTPS
- label should not contain HTML: because of the escaped &

I'm going to look at it

### PR DESCRIPTION
**Pull request by virtualtam**
This is a rebased and updated version of #919 by @Lucas-C with updated test code

Closes #914
Closes #919

### CODE REVIEW

**Comment by ArthurHoaro** [comment]
This doesn't work because the "Permalink" link isn't escaped. It also doesn't work with markdown, because the plugin hook is called after the feed data generation, and it creates non escaped HTML tags.

A solution would be to escape the content out of `FeedBuilder` and after calling the hooks. But IMHO removing the `CDATA` wrapper here is unnecessary.

**Comment by virtualtam** [comment]
Good idea, the data flow should look like this:
- prepare feed data
- apply plugin hooks
- sanitize, escape, etc.
- render
- cache

It'd be worth refactoring the related components to simplify feed generation: `FeedBuilder`, `Cache`, `CachedPage`

File: application/FeedBuilder.php [hunk 1]
```diff
@@ -153,6 +153,7 @@ protected function buildItem($link, $pageaddr)
             $permalink = '<a href="'. $link['guid'] .'" title="'. t('Permalink') .'">'. t('Permalink') .'</a>';
         }
         $link['description']  = format_description($link['description'], '', false, $pageaddr);
+        $link['description'] = $this->htmlEscape($link['description']);
```
Comment (ArthurHoaro) [inline application/FeedBuilder.php]: Any reason not to use `escape()`?

**Comment by ArthurHoaro** [comment]
Closing due to inactivity/being out of date.

**Pull request closed without merging**